compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1165,6 +1169,12 @@ impl Target {
                     base.options.$key_name = s;
                 }
             } );
+            ($key_name:ident, Option<u32>) => ( {
+                let name = (stringify!($key_name)).replace("_", "-");
+                if let Some(s) = obj.find(&name).and_then(Json::as_u64) {
+                    base.options.$key_name = Some(s as u32);
```

Comment:
We need to produce an error here by returning `Err(String)` if `"dwarf-version"` is not an integer in the range 1-5.